blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineImpl.java

```diff
@@ -200,5 +204,19 @@ public BluePipeline getPipeline(Item item, Reachable parent) {
         return actionProxies;
 
     }
+    
+    // TODO: @vivek/@ivan please fix me the right way !!!
+    public static @Nonnull Link getLink(@Nonnull ParameterizedJobMixIn.ParameterizedJob job) {
```

Comment:
I put my comments on BlueMessageEnricher. Need to understand uses of it bit better to further comment on how it works.